contentscript.js

```diff
@@ -89,10 +97,46 @@
                         alternateProfile: request.data.alternateProfile
                     }, function() {});
                 }
+
+                if ('enableZip' in request.data) {
+                    chrome.storage.sync.set({
+                        enableZip: request.data.enableZip
+                    }, function() {});
+                    setZipEnabled(!!request.data.enableZip);
+                }
+
             }
         );
     }
 
+    /**
+     * Set whether or not ZIP support is enabled.
+     */
+    function setZipEnabled(enable) {
+        setWindowVariable('enableZip', enable);
+
+        if(!enable) {   // Disable: hide the ZIP controls.
+            // On the shader page
+            var elem = document.getElementById('dl-shader-zip');
+            if(elem) elem.style.display = 'none';
+
+            // On the profile page
+            elem = document.getElementById('dl-all-shader-zip');
+            if(elem) elem.style.display = 'none';
+
+        } else {        // Enable
+            window.loadZipScripts();
+
+            // Enable button on the shader page
+            var elem = document.getElementById('dl-shader-zip');
+            if(elem) elem.style.display = 'block';
+
+            // Enable button on the profile page
+            elem = document.getElementById('dl-all-shader-zip');
+            if(elem) elem.style.display = 'block';

```

Comment:
IFIK this PR is about downloading ZIP with single shader :)

Reply:
That is a good point.  However, I don't want to implement this in a way that won't also work for the profile page.  I will test both, then remove the profile-page material when I send you the complete version for your review.  That will make it easy to finish #49 after this is merged.